pkg/client/client.go

```diff
@@ -0,0 +1,232 @@
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", socket)
+			},
+		},
+	}
+
+	req, err := http.NewRequest(method, "http://localhost"+path, bytes.NewReader([]byte(body)))
+	if err != nil {
+		return nil, err
+	}
+
+	for h, v := range headers {
+		req.Header.Set(h, v)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+// APIErrorMsg is an individual API error with an ID and a message string
+type APIErrorMsg struct {
+	ID  string `json:"id"`
+	Msg string `json:"msg"`
+}
+
+// String returns the error id and message as a string
+func (r *APIErrorMsg) String() string {
+	return fmt.Sprintf("%s: %s", r.ID, r.Msg)
+}
+
+// APIResponse is returned with status code 400 and may contain a list of errors
+// If Status is true the Errors list will not be included or will be empty.
+// When Status is false it will include at least one APIErrorMsg with details about the error.
+// It also implements the error interface so that it can be used in place of error
+type APIResponse struct {
+	Status bool          `json:"status"`
+	Errors []APIErrorMsg `json:"errors,omitempty"`
+}
+
+// Error returns the description of the first error
+func (r *APIResponse) Error() string {
+	if len(r.Errors) == 0 {
+		return ""
+	}
+	return r.Errors[0].String()
+}
+
+// AllErrors returns a list of error description strings
+func (r *APIResponse) AllErrors() (all []string) {
+	for i := range r.Errors {
+		all = append(all, r.Errors[i].String())
+	}
+	return all
+}
+
+// clientError converts an error into an APIResponse with ID set to ClientError
+// This is used to return golang function errors to callers of the client functions
+func clientError(err error) *APIResponse {
+	return &APIResponse{
+		Status: false,
+		Errors: []APIErrorMsg{{ID: "ClientError", Msg: err.Error()}},
+	}
+}
+
+// NewAPIResponse converts the response body to a status response
+func NewAPIResponse(body []byte) *APIResponse {
+	var status APIResponse
+	err := json.Unmarshal(body, &status)
+	if err != nil {
+		return clientError(err)
+	}
+	return &status
+}
+
+// apiError converts an API error 400 JSON to a status response
+//
+// The response body should alway be of the form:
+//     {"status": false, "errors": [{"id": ERROR_ID, "msg": ERROR_MESSAGE}, ...]}
+func apiError(resp *http.Response) *APIResponse {
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return clientError(err)
+	}
+	return NewAPIResponse(body)
+}
+
+// GetRaw returns raw data from a GET request
+// Errors from the client and from the API are returned as an APIResponse
+func GetRaw(socket, method, path string) ([]byte, *APIResponse) {
+	resp, err := Request(socket, method, path, "", map[string]string{})
+	if err != nil {
+		return nil, clientError(err)
+	}
+
+	// Convert the API's JSON error response to an error type and return it
+	if resp.StatusCode == 400 {
```

Comment:
The caller of the `GetRaw` function can distinguish if the status code was 400 or not. Is it enough? We might also want to test other status codes.

Reply:
Error 400 is specific to the API responses, it indicates that the body should be a JSON API error response. All other errors, like 404, should be handled by the caller.